### ISSUE TITLE
JSON Error when reading message

### ISSUE THREAD

**Issue by Dankrushen** [DV8FromTheWorld/JDA#568]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

Hey there! Before you report a bug or suggest a new feature,
  please make sure to follow these steps first!
  
- [x] I have checked for similar issues.
Possibly similar issue: https://github.com/DV8FromTheWorld/JDA/issues/304
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming features/bug fixes.

> This is not the place to learn Java. Please refer to [StackOverflow][stack overflow]
  for your general programming questions.

### Issue Type

- [x] Bug Report
- [ ] Feature Request

### Description

I continuously get an error when my bot attempts to read this one message on a server because Discord is sending incorrect JSON data, and I want some way to skip over the message when iterating through a channel's IterableHistory like so:
```java
MessagePaginationAction paginator = channel.getIterableHistory();
paginator.cache(false); 

Iterator<Message> iterator = paginator.iterator();

while(iterator.hasNext()) {
    Message message = iterator.next();
}
```

Here's the error the code above produces when it iterates over the specific message:
```
INFO 15:30:30.132 [JDA RateLimit-Queue Pool - Thread 4] ERROR n.d.jda.core.requests.Requester - There was an exception while executing a REST request
INFO org.json.JSONException: Encountered embed object with missing/null type field for Json: {"provider":{"name":"Nocookie","url":"http://vignette4.wikia.nocookie.net"},"type":null,"url":"http://vignette4.wikia.nocookie.net/warframe/images/4/4f/TylRegorBattleTaunt2.ogg/revision/latest?cb=20150801042748"}
```

**Comment by MinnDevelopment** [comment]
This will be possible once #453 has been merged.

**Comment by DV8FromTheWorld** [comment]
Since #453 has been merged, can this be closed?

**Comment by MinnDevelopment** [comment]
I want to close issues related to changes once those changes are released on master

**Comment by MinnDevelopment** [comment]
Fixed in [v3.4.0](https://github.com/DV8FromTheWorld/JDA/releases/tag/v3.4.0)
